### PR TITLE
Add /timecard feature to develop branch

### PR DESCRIPTION
**Pull request by jonvaljonathan**
Add the /timecard feature's core functionality.

/timecard checkin
/timecard checkout
/timecard hours

### CODE REVIEW

**Comment by amaredeus** [comment]
@jonvaljonathan Looks like your forked repo's develop branch is behind the org's develop branch. Can you try syncing it? https://github.com/jonvaljonathan/discord-bots/tree/develop there might be a button to do this easily on the github page for your repo

File: src/app/schema/timecard/timecard.json [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        "startTime": {
+          "bsonType": ["string", "null"],
+          "description": "ISO8601 of when the meeting started."
+        },
+        "endTime": {
+          "bsonType": ["string", "null"],
+          "description": "ISO8601 of when the meeting ended."
+        },
+        "duration": {
+          "bsonType": ["number", null],
```
Comment (amaredeus) [inline src/app/schema/timecard/timecard.json:27]: null should be enclosed in double quotations

File: src/app/types/timecard.ts/Timecard.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+export interface Timecard extends Collection {
+	_id: ObjectId,
+	startTime: string,
+	endTime: string,
+    description: string,
+    duration: number,
+	discordUserId: string,
+	discordServerId: string,
+    isActive: boolean,
+}
```
[the remark this answers is not in the document]
Comment (amaredeus) [reply src/app/types/timecard.ts/Timecard.ts:12]: yeah the linter doesn't seem to capture typescript types or interfaces 🤷

**Review by amaredeus — CHANGES_REQUESTED** [review]
PR is starting to shape up very well! Can we also jot down the change under CHANGELOG.md? Once that's down and the other comments are addressed, I think this PR will be ready for this week's release!

**Comment by amaredeus** [comment]
Hmm it looks like the test cases are failing https://github.com/jonvaljonathan/discord-bots/runs/4092653720?check_suite_focus=true

If you run `yarn test` what do you see?

**Review by amaredeus — DISMISSED** [review]

**Review by nonsensecodes — APPROVED** [review]

**Pull request merged**